crates/turbo-tasks-memory/src/task_statistics.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::{Arc, OnceLock};
+
+use dashmap::DashMap;
+use serde::{ser::SerializeMap, Serialize, Serializer};
+use turbo_tasks::{registry, FunctionId};
+
+/// An API for optionally enabling, updating, and reading aggregated statistics.
+#[derive(Default)]
+pub struct TaskStatisticsApi {
+    inner: OnceLock<Arc<TaskStatistics>>,
+}
+
+impl TaskStatisticsApi {
+    pub fn enable(&self) -> &Arc<TaskStatistics> {
+        // ignore error: potentially already initialized, that's okay
+        let _ = self.inner.set(Arc::new(TaskStatistics {
+            inner: DashMap::new(),
+        }));
```

Comment:
I think we should use [`get_or_init`](https://doc.rust-lang.org/std/sync/struct.OnceLock.html#method.get_or_init) instead